compiler/damlc/tests/BUILD.bazel

```diff
@@ -107,11 +107,80 @@ da_haskell_test(
     ],
 )
 
-# Integration tests for all non-dev DAML-LF versions
-damlc_integration_test("integration-stable", "DA.Test.DamlcIntegration.mainAll")
+da_haskell_library(
+    name = "integration-lib",
+    srcs = ["src/DA/Test/DamlcIntegration.hs"],
+    deps = [
+        "//compiler/daml-lf-ast",
+        "//compiler/daml-lf-proto",
+        "//compiler/damlc/daml-compiler",
+        "//compiler/damlc/daml-ide-core",
+        "//compiler/damlc/daml-lf-conversion",
+        "//compiler/damlc/daml-opts:daml-opts-types",
+        "//compiler/damlc/daml-opts",
+        "//compiler/scenario-service/client",
+        "//daml-lf/archive:daml_lf_dev_archive_haskell_proto",
+        "//libs-haskell/bazel-runfiles",
+        "//libs-haskell/da-hs-base",
+        "//libs-haskell/test-utils",
+    ],
+    hackage_deps = [
+        "aeson-pretty",
+        "base",
+        "bytestring",
+        "data-default",
+        "deepseq",
+        "directory",
+        "dlist",
+        "extra",
+        "filepath",
+        "ghc-lib",
+        "ghc-lib-parser",
+        "ghcide",
+        "haskell-lsp-types",
+        "optparse-applicative",
+        "process",
+        "proto3-suite",
+        "shake",
+        "tagged",
+        "tasty",
+        "tasty-hunit",
+        "text",
+        "time",
+        "unordered-containers",
+    ],
+    visibility = ["//visibility:public"],
+)
 
-# Integration tests for DAML-LF 1.dev
-damlc_integration_test("integration-dev", "DA.Test.DamlcIntegration.main")
+[
+    da_haskell_test(
```

Comment:
I wonder if it makes sense to make `integration-lib` above a binary this one a `sh_test`. Any thoughts?

Reply:
In principle this should work, however, there is an issue with the path to the runfiles tree not being overridden correctly when a Haskell binary is wrapped in a `sh_test` directly. In this particular case none of the `data` dependencies on the test case are specific to the LF version, so moving the `data` attribute to the underlying `da_haskell_binary` should work around that issue.

Reply:
I'll try that in a follow-up PR. Thanks.